closes #11261 -  Inject document separators when rendering CRDs

KyleFrisbie opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. Make sure to read the Contributing Guide before submitting your PR: https://github.com/helm/helm/blob/main/CONTRIBUTING.md
2. If this PR closes another issue, add 'closes #<issue number>' somewhere in the PR summary. GitHub will automatically close that issue when this PR gets merged. Alternatively, adding 'refs #<issue number>' will not close the issue, but help provide the reviewer more context.-->

**What this PR does / why we need it**:
This PR address issue #11261 and accounts for charts with CRDs that don't contain file separators at the beginning of the file, as mentioned in the issue.
**Special notes for your reviewer**:
closes #11261
**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

Signed-off-by: Kyle L Frisbie <kyle.l.frisbie@gmail.com>

pscheid92 committed:
feature(helm): add --set-literal flag for literal string interpretation

The current family of '--set' methods interprets some special characters
in values (e.g. commas, square brackets, points, backslashes). With the
typical shell escaping rules, this can increase the difficulty of overwriting
values in some cases.

In contrast to '--set-string' or similar methods, '--set-literal' does
not interpret those special characters. It interprets given values as
literal strings.

Example:

    --set-literal outer.inner='so\me,values'

    outer:
      inner: so\me,values

Closes #4030

Signed-off-by: Patrick Scheid <p.scheid92@gmail.com>

programmer04 committed:
Fix User-Agent header in requests made by Helm

Signed-off-by: Jakub Warczarek <jakub.warczarek@gmail.com>

programmer04 committed:
Add test for User-Agent header setting and refactor

Signed-off-by: Jakub Warczarek <jakub.warczarek@gmail.com>

SoujanyaMangipudi committed:
feat(helm): Supporting helm3 to show up resource names that were deployed as part of release in helm status command

Creating a new PR based on this existing stale PR https://github.com/helm/helm/pull/7728

Signed-off-by: Soujanya Mangipudi <somangip@microsoft.com>

# Conflicts:
#	go.sum

SoujanyaMangipudi committed:
Addressing review comments:
Extend Interface with new InterfaceResources to avoid breaking changes
Move change to staus command behind --show-resources flag

Signed-off-by: Soujanya Mangipudi <somangip@microsoft.com>

souleb committed:
Enable custom certificates option for OCI

If implemented, users will be able to use custom certificates and CA to
while interacting with OCI registries.

Signed-off-by: Soule BA <bah.soule@gmail.com>

porridge committed:
During deletion, explicitly log already deleted resource name.

Unfortunately errors from the API server do not always (do they ever?) contain
the name of the resource in question.

Deletions for multiple resources are processed concurrently, so in a resulting
log, a preceding "Starting delete" line might be for a different object.

Signed-off-by: Marcin Owsiany <porridge@redhat.com>

Commit:
fix: clean up temp files in FindChartInAuthAndTLSAndPassRepoURL (#11171)

Signed-off-by: CI <ci@argoproj.com>

Commit:
fix tests

Signed-off-by: CI <ci@argoproj.com>

Commit:
avoid adding new public function

Signed-off-by: CI <ci@argoproj.com>

Commit:
don't change r.CachePath

Signed-off-by: CI <ci@argoproj.com>

Commit:
one defer

Signed-off-by: CI <ci@argoproj.com>

Commit:
Merge remote-tracking branch 'origin/main' into cleanup-after-find-chart

SoujanyaMangipudi committed:
Addressing review comments - move printing code out of client.go

Signed-off-by: Soujanya Mangipudi <somangip@microsoft.com>

fix circleci tests

Signed-off-by: Soujanya Mangipudi <somangip@microsoft.com>

fix formatting errors

Signed-off-by: Soujanya Mangipudi <somangip@microsoft.com>

fix tests

Signed-off-by: Soujanya Mangipudi <somangip@microsoft.com>

fix tests

Signed-off-by: Soujanya Mangipudi <somangip@microsoft.com>

fix gofmt issue

Signed-off-by: Soujanya Mangipudi <somangip@microsoft.com>

SoujanyaMangipudi committed:
Merge pull request #3 from SoujanyaMangipudi/somangip/MergetoMain

Somangip/mergeto main

SoujanyaMangipudi committed:
Merge pull request #3 from SoujanyaMangipudi/somangip/MergetoMain

Somangip/mergeto main

Signed-off-by: Soujanya Mangipudi <somangip@microsoft.com>

SoujanyaMangipudi committed:
Merge branch 'main' of https://github.com/SoujanyaMangipudi/helm into main

joejulian committed:
Allow CGO_ENABLED to be overridden for build

On Mac OS they have some custom dns c library that uses some
configuration files other than resolv.conf to configure dns lookups. The
standard go library does not handle these custom configuration files
which causes dns lookups to fail for some mac users.

This allows the downstream pacakgers to override CGO_ENABLED to build
binaries that use the custom dns library.

Signed-off-by: Joe Julian <me@joejulian.name>
Signed-off-by: Kyle L Frisbie <kyle.l.frisbie@gmail.com>

KyleFrisbie committed:
close #11261 -  Inject document separators when rendering CRDs

Signed-off-by: Kyle L Frisbie <kyle.l.frisbie@gmail.com>

KyleFrisbie committed:
close #11261 -  Inject document separators when rendering CRDs

Signed-off-by: Kyle L Frisbie <kyle.l.frisbie@gmail.com>

cenkalti committed:
retry http request on temporary errors

Signed-off-by: Cenk Alti <cenkalti@gmail.com>

mattfarina committed:
Updating to kubernetes 1.25.2 packages

Signed-off-by: Matt Farina <matt.farina@suse.com>

mattfarina committed:
Merge pull request #11172 from crenshaw-dev/cleanup-after-find-chart

fix: clean up temp files in FindChartInAuthAndTLSAndPassRepoURL (#11171)

mattfarina committed:
Merge pull request #11427 from mattfarina/bump-0.25.2

Updating to kubernetes 1.25.2 packages

mattfarina committed:
Updating the repo the azure cli is installed from

CircleCI is now using Ubuntu 20.04 for the Go 1.18 image. The
azure CLI is expecting a version older than that. This change
updates.

Note, we have been using the debian name for the repo rather than
the ubuntu name. bullseye maps to Ubuntu 20.04 through 21.10.

Signed-off-by: Matt Farina <matt.farina@suse.com>

mattfarina committed:
Merge pull request #11433 from mattfarina/fix-deploy

cenkalti committed:
Revert "Tolerate temporary errors from etcdserver"

This reverts commit ebc79fa807f29b984e090f0071b640f7347937cf.

Signed-off-by: Cenk Alti <cenkalti@gmail.com>

mattfarina committed:
Updating the deb location for azure cli

This update moves to ubuntu jammy. The name moves off of debian
version names because jammy is based on bookworm and azure cli
does not have debian package support with bookworm

This is an update to the previous update for this because the
CircleCI environment is ubuntu 20.04 but the container this is
running is has 22.04 as the ubuntu version. Updating the in
container to 22.04 (jammy).

Signed-off-by: Matt Farina <matt.farina@suse.com>

sabre1041 committed:
Merge pull request #11436 from mattfarina/fix-deploy2

Updating the deb location for azure cli

phoenixsqf committed:
use intstr.GetScaledValueFromIntOrPercent instead of the deprecated

Signed-off-by: Qifan Shen <sqfphoenix@163.com>

cuishuang committed:
fix a few function names on comments

Signed-off-by: cui fliter <imcusg@gmail.com>

hickeyma committed:
Update schema validation handling

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

p33t committed:
Readiness & liveness probes correct port

Port number now comes from values.yaml

Signed-off-by: Peter Leong <1179301+p33t@users.noreply.github.com>

hickeyma committed:
Merge pull request #11462 from p33t/main

Readiness & liveness probes correct port

dependabot[bot] committed:
chore(deps): bump github.com/spf13/cobra from 1.5.0 to 1.6.1

Bumps [github.com/spf13/cobra](https://github.com/spf13/cobra) from 1.5.0 to 1.6.1.
- [Release notes](https://github.com/spf13/cobra/releases)
- [Commits](https://github.com/spf13/cobra/compare/v1.5.0...v1.6.1)

---
updated-dependencies:
- dependency-name: github.com/spf13/cobra
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

MichaelMorrisEst committed:
Add option to support cascade deletion options

Add --cascade=<background|foreground|orphan> option to helm uninstall
Current behaviour is hardcoded to background

Addresses issue: https://github.com/helm/helm/issues/10586

Signed-off-by: MichaelMorris <michael.morris@est.tech>

jouve committed:
redirect registry client output to stderr

Signed-off-by: Cyril Jouve <jv.cyril@gmail.com>

yxxhero committed:
fix template --output-dir issue

Signed-off-by: yxxhero <aiopsclub@163.com>

Commit:
update .golangci for go1.18

Signed-off-by: yanggang <gang.yang@daocloud.io>

learnitall committed:
Add support for CSVs in template --api-versions arg

This commit changes the template sub-command argument
`--api-versions` from `StringArrayVarP` to `StringSliceVarP`,
to allow support for comma separated variables. This will allow
users to use a syntax such as `--api-versions one,two,three` over
`--api-versions one --api-versions two --api-versions three`.

Fixes: #11485

Signed-off-by: Ryan Drew <ryan.drew@isovalent.com>

sabre1041 committed:
Merge pull request #11300 from porridge/better-logging

During deletion, explicitly log already deleted resource.

Wonong committed:
Add a flag var to check git is installed or not

Signed-off-by: Wonyeong Choi <ciwnyg0815@gmail.com>

phil9909 committed:
improve error message on plugin install

Signed-off-by: Philipp Stehle <philipp.stehle@sap.com>

sabre1041 committed:
Merge pull request #11480 from jouve/oci

redirect registry client output to stderr

Wonong committed:
Update logic of non-git situation just to print warning logs

Signed-off-by: Wonyeong Choi <ciwnyg0815@gmail.com>

mattfarina committed:
Merge pull request #11442 from catandcoder/main

fix a few function names on comments

phil9909 committed:
harmonize URL reference resolving

Fixes https://github.com/helm/helm/issues/11472

Signed-off-by: Philipp Stehle <philipp.stehle@sap.com>

hickeyma committed:
Update repo handling

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

hickeyma committed:
Merge pull request #11426 from cenkalti/leader-changed

Fix handling of "leader changed" errors

sabre1041 committed:
Added support for insecure OCI registries

Signed-off-by: Andrew Block <andy.block@gmail.com>

gjenkins8 commented on pkg/action/show_test.go:
```diff
@@ -102,6 +102,7 @@ func TestShowCRDs(t *testing.T) {
 			{Name: "crds/ignoreme.txt", Data: []byte("error")},
 			{Name: "crds/foo.yaml", Data: []byte("---\nfoo\n")},
 			{Name: "crds/bar.json", Data: []byte("---\nbar\n")},
+			{Name: "crds/baz.yaml", Data: []byte("\nbaz\n")},
```
nit: why the leading newline?

```suggestion
			{Name: "crds/baz.yaml", Data: []byte("baz\n")},
```

hickeyma committed:
Update string handling

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

hickeyma committed:
Merge pull request #11494 from Wonong/add-check-git

Add a flag var to check git is installed or not

yxxhero committed:
bump sprig version 3.2.3

Signed-off-by: yxxhero <aiopsclub@163.com>

hickeyma committed:
Merge pull request #11582 from yxxhero/bump-sprig-version-3.2.3

bump sprig version 3.2.3

dependabot[bot] committed:
chore(deps): bump golang.org/x/text from 0.4.0 to 0.5.0

Bumps [golang.org/x/text](https://github.com/golang/text) from 0.4.0 to 0.5.0.
- [Release notes](https://github.com/golang/text/releases)
- [Commits](https://github.com/golang/text/compare/v0.4.0...v0.5.0)

---
updated-dependencies:
- dependency-name: golang.org/x/text
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Commit:
Update chartrepo.go

## case :
1. charmuseum is http
2. nginx https ---> chartmuseum
3. domain, CN is IP

## result:  
 error: failed to prepare chart for release: chart unavailable: looks like "https://120.55.54.50" is not a valid chart repository or cannot be reached: Get "https://120.55.54.50/index.yaml": x509: certificate relies on legacy Common Name field, use SANs instead



Signed-off-by: caixisheng <cc710917049@163.com>

lucacome committed:
Bump containerd to 1.6.15, oras-go to 1.2.2 and image-spec to v1.1.0-rc2

Signed-off-by: Luca Comellini <luca.com@gmail.com>

Commit:
docs: add docs for cli/values.Options

Signed-off-by: Zuhair AlSader <zuhair@koor.tech>

hickeyma committed:
Fix backwards compatibility

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

hickeyma committed:
Merge pull request #11508 from sap-contributions/improve-plugin-install-error

improve error message on plugin install

sabre1041 committed:
Added insecure option to login subcommand

Signed-off-by: Andrew Block <andy.block@gmail.com>

lucacome committed:
Bump k8s.io deps to v0.26.0

Signed-off-by: Luca Comellini <luca.com@gmail.com>

mattfarina committed:
Merge pull request #11470 from helm/dependabot/go_modules/github.com/spf13/cobra-1.6.1

chore(deps): bump github.com/spf13/cobra from 1.5.0 to 1.6.1

dastrobu committed:
change linting error messages for null values in arrays

Closes #11627

Signed-off-by: Daniel Strobusch <1847260+dastrobu@users.noreply.github.com>

sabre1041 committed:
Removed conditional

Signed-off-by: Andrew Block <andy.block@gmail.com>

programmer04 committed:
Trigger CI

Signed-off-by: Jakub Warczarek <jakub.warczarek@gmail.com>

sabre1041 committed:
Reimplemented change in httpgetter for insecure TLS option

Signed-off-by: Andrew Block <andy.block@gmail.com>

SoujanyaMangipudi committed:
Merge branch 'main' into main

Signed-off-by: Soujanya Mangipudi <somangip@microsoft.com>

SoujanyaMangipudi committed:
Resolve conflicts for go.mod and go.sum

Signed-off-by: Soujanya Mangipudi <somangip@microsoft.com>

hickeyma committed:
Merge pull request from GHSA-67fx-wx78-jx33

Update schema validation handling

hickeyma committed:
Merge pull request from GHSA-53c4-hhmh-vw5q

Update repo handling

hickeyma committed:
Merge pull request from GHSA-6rx9-889q-vv2r

Update string value name handling

mattfarina committed:
Merge pull request #11601 from helm/dependabot/go_modules/golang.org/x/text-0.5.0

chore(deps): bump golang.org/x/text from 0.4.0 to 0.5.0

souleb committed:
Provide a helper to set the registryClient in cmd

If enabled the registryClient is set using a helper that accepts the TLS
flags. This keeps the client creation consistent accross the different
commands.

Signed-off-by: Soule BA <bah.soule@gmail.com>

mattfarina committed:
Merge pull request #11604 from xishengcai/patch-1

Update chartrepo.go

SoujanyaMangipudi committed:
Merge remote-tracking branch 'helm-upstream/main' into somangip/MergeHelmMain1219

Signed-off-by: Soujanya Mangipudi <somangip@microsoft.com>

SoujanyaMangipudi committed:
Merge pull request #5 from SoujanyaMangipudi/somangip/MergeHelmMain1219

Somangip/merge helm main1219

programmer04 committed:
Fix after CR

Signed-off-by: Jakub Warczarek <jakub.warczarek@gmail.com>

mattfarina committed:
Merge pull request #10912 from SoujanyaMangipudi/main

feat(helm): Supporting helm3 to show up resource names in helm status

dependabot[bot] committed:
chore(deps): bump github.com/BurntSushi/toml from 1.2.0 to 1.2.1

Bumps [github.com/BurntSushi/toml](https://github.com/BurntSushi/toml) from 1.2.0 to 1.2.1.
- [Release notes](https://github.com/BurntSushi/toml/releases)
- [Commits](https://github.com/BurntSushi/toml/compare/v1.2.0...v1.2.1)

---
updated-dependencies:
- dependency-name: github.com/BurntSushi/toml
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

VaibhavSharma-47 committed:
fix adopted resource not replaced

Signed-off-by: Vaibhav Sharma <17532va@gmail.com>

sabre1041 committed:
CircleCI Migration

Signed-off-by: Andrew Block <andy.block@gmail.com>

mattfarina committed:
Merge pull request #11616 from zalsader/add-docs-cli-values

docs: add docs for cli/values.Options

mattfarina committed:
Merge pull request #11524 from sap-contributions/harmonize-url-reference-resolving

harmonize URL reference resolving

mattfarina committed:
Merge branch 'main' into 4030-provide-literal-alternative-for-set-flag

Signed-off-by: Matt Farina <matt@mattfarina.com>

mattfarina committed:
Fixing indentation

Signed-off-by: Matt Farina <matt.farina@suse.com>

hickeyma committed:
Merge pull request #11226 from programmer04/fix-user-agent

Fix User-Agent header in requests made by Helm

hickeyma committed:
Merge pull request #11622 from lucacome/bump-k8s-deps

Bump k8s.io deps to v0.26.0

mattfarina committed:
Merge pull request #11545 from helm/dependabot/go_modules/github.com/BurntSushi/toml-1.2.1

chore(deps): bump github.com/BurntSushi/toml from 1.2.0 to 1.2.1

hickeyma committed:
Merge pull request #11628 from dastrobu/Linting_error_on_null_in_dependencies_is_misleading_#11627

change linting error messages for null values in arrays

pscheid92 committed:
Add protection for stack-overflows for nested keys

Signed-off-by: Patrick Scheid <p.scheid92@gmail.com>

mattfarina committed:
Merge pull request #11609 from lucacome/bump-containerd

Bump containerd to 1.6.15, oras-go to 1.2.2 and image-spec to v1.1.0-rc2

mattfarina committed:
Merge pull request #11489 from yanggangtony/update-go-lint

update .golangci for go1.18

mattfarina committed:
Merge pull request #11490 from learnitall/pr/learnitall/allow-csv-api-versions

Add support for CSVs in template --api-versions arg

mattfarina committed:
Merge pull request #11439 from phoenixsqf/main

use intstr.GetScaledValueFromIntOrPercent instead of the deprecated

mattfarina committed:
Merge pull request #11672 from VaibhavSharma-47/feature

[CMD/PKG] Adopted resource not replaced when using upgrade --install --force

mattfarina committed:
bump version to v3.11.0

Signed-off-by: Matt Farina <matt.farina@suse.com>
(cherry picked from commit 9d8fee155bd7e7d3c1390f4076d9271a1147dce5)

mattfarina committed:
Merge pull request #11704 from mattfarina/bump-version-3.11

bump version to v3.11.0

mattfarina committed:
Fix improper use of Table request/response to k8s API

Fixes #11712

A change was made that when validation was turned off the Kubernetes
packages were building objects as a Table type. This was done for
display purposes. When details about the objects was going to be
printed as part of #10912.

This broke rollback, and possibly other functionality, as a Table
type was returned in some cases that needed the regular object.
This caused things to break silently.

The fix involved adding in a new Function (and interface) to
query for tables instead of the objects themselves. There was not
a clean way to add it to the existing function that covered all
cases.

A second problem was noticed along the way. When data was output
via status as YAML or JSON it was in the form of a table rather
than the objects themselves. This did not reflect expectations
and did not match the functionality in kubectl. The code was
updated to return a table when that was presented and the objects
when they are being output for YAML or JSON. The API also supports
this handling to SDK users can replicate this functionality.

API changes made here were never released. The functions were
developed for this release of Helm and only ever appeared in an
RC. In this case, they can be changed.

Signed-off-by: Matt Farina <matt.farina@suse.com>

cenkalti committed:
Check status code before retrying request

Signed-off-by: Cenk Alti <cenkalti@gmail.com>

mattfarina committed:
Merge pull request #11723 from cenkalti/leader-changed-fix

Check status code before retrying request

mattfarina committed:
Merge pull request #11720 from mattfarina/fix-11712

Fix improper use of Table request/response to k8s API

Commit:
the linter varcheck and deadcode are deprecated (since v1.49.0)

Signed-off-by: Fish-pro <zechun.chen@daocloud.io>

mattfarina committed:
Update to func handling

Signed-off-by: Matt Farina <matt@mattfarina.com>

mattfarina committed:
Merge pull request #11487 from yxxhero/fix_template_with_output_dir

fix template --output-dir issue

dependabot[bot] committed:
chore(deps): bump github.com/rubenv/sql-migrate from 1.2.0 to 1.3.1

Bumps [github.com/rubenv/sql-migrate](https://github.com/rubenv/sql-migrate) from 1.2.0 to 1.3.1.
- [Release notes](https://github.com/rubenv/sql-migrate/releases)
- [Commits](https://github.com/rubenv/sql-migrate/compare/v1.2.0...v1.3.1)

---
updated-dependencies:
- dependency-name: github.com/rubenv/sql-migrate
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

mattfarina committed:
Merge pull request from GHSA-pwcw-6f5g-gxf8

Update to func handling

joejulian committed:
don't mark issues as stale where a PR is in progress

Signed-off-by: Joe Julian <me@joejulian.name>

hickeyma committed:
Merge pull request #11806 from joejulian/in_progress

don't mark issues as stale where a PR is in progress

hickeyma committed:
Merge pull request #11779 from Fish-pro/chore/golangci

chore(golang-ci): the linter `varcheck` and `deadcode` are deprecated (since v1.49.0)

scottrigby committed:
Convert remaining CircleCI config to GitHub Actions

Signed-off-by: Scott Rigby <scott@r6by.com>

cpanato committed:
update k8s registry domain

Signed-off-by: cpanato <ctadeu@gmail.com>

hickeyma committed:
Merge pull request #11829 from cpanato/renamek8s

update k8s registry domain

my-git9 committed:
chore:Use http constants as http.request parameters

Signed-off-by: xin.li <xin.li@daocloud.io>

sabre1041 committed:
Changed how the setup-go action sets go version

Signed-off-by: Andrew Block <andy.block@gmail.com>

scottrigby committed:
Add why comments

Signed-off-by: Scott Rigby <scott@r6by.com>

scottrigby committed:
Remove check to run only in helm org

Signed-off-by: Scott Rigby <scott@r6by.com>

scottrigby committed:
Remove job dependency. Should have done when I moved job to new file

Signed-off-by: Scott Rigby <scott@r6by.com>

hickeyma committed:
Merge pull request #11799 from helm/dependabot/go_modules/github.com/rubenv/sql-migrate-1.3.1

chore(deps): bump github.com/rubenv/sql-migrate from 1.2.0 to 1.3.1

sabre1041 committed:
Merge pull request #2 from scottrigby/merge-11820-into-11676-clean

Merge 11820 into 11676

sabre1041 committed:
Removed remaining CircleCI artifacts

Signed-off-by: Andrew Block <andy.block@gmail.com>

sabre1041 committed:
Updates based on feedback

Signed-off-by: Andrew Block <andy.block@gmail.com>

sabre1041 committed:
Added back CircleCI config.yaml

Signed-off-by: Andrew Block <andy.block@gmail.com>

sabre1041 committed:
Merge pull request #11676 from sabre1041/circleci-migration

CircleCI Migration

mattfarina committed:
Add testing to default and release branches

Also added a note to the circleci config

Signed-off-by: Matt Farina <matt.farina@suse.com>

mattfarina committed:
Merge pull request #11865 from mattfarina/gha-fixup

Add testing to default and release branches

cbodonnell committed:
full source path when output-dir is not provided

Signed-off-by: Craig O'Donnell <craig@replicated.com>

hiddeco committed:
Properly invalidate client after CRD install

As the CRDs are installed before the capabilities are gathered, the
current call to invalidate the discovery client is premature and
expensive.

What actually is required is an invalidation of the REST mapper, as
otherwise the Helm install action may later on fail with a `resource
mapping not found` error. More specifically when the caller of the
action is making use of a persisting[1] `RESTClientGetter`.

Which is not something done by the Helm CLI (albeit it could, and this
would potentially save quite some resources?). But is a default
configuration offered by the Helm SDK via `kube.New` when a nil value
is provided as the `getter`.

[1]: https://github.com/kubernetes/cli-runtime/blob/v0.26.2/pkg/genericclioptions/config_flags.go#L118

Signed-off-by: Hidde Beydals <hidde@hhh.computer>

mattfarina committed:
Bump the Go version

Needed to gofmt source to meet changes in style

Signed-off-by: Matt Farina <matt.farina@suse.com>

Commit:
Fix goroutine leak in perform

Signed-off-by: willzgli <willzgli@tencent.com>

scottrigby committed:
Merge pull request #11711 from helm/oci-transport-refinement

mattfarina committed:
Updating the Go version in go.mod

At this time both Go 1.19 and 1.20 are supported. The version
specified in the go.mod file is the minimum version we expect Helm
to be compiled against. This is the oldest supported version to
support environments where others compile Helm. The Helm project
is using Go 1.20 to build Helm itself.

Updating to Go 1.19 also includes dealing with io/ioutil
deprecation and some additional linting issues around staticcheck.
All the staticcheck issues were in test files so linting was
skipped for those.

Signed-off-by: Matt Farina <matt.farina@suse.com>

mattfarina committed:
Merge pull request #11886 from hiddeco/properly-invalidate-client

Properly invalidate client after CRD install

mattfarina committed:
Handle failed DNS case for Go 1.20+

Go 1.20 introduced DNS/CNAME handling changes. This can cause an
IP address represented in hex, oct, etc to be looked up as DNS and
fail. This change introduces a mock DNS resolver.

Note, with the mock resolver, we don't need to use 0x7f000001 any
longer. Keeping because it was already there.

Signed-off-by: Matt Farina <matt.farina@suse.com>

mattfarina committed:
Merge pull request #11896 from mattfarina/go-1.20

Bump the Go version

dependabot[bot] committed:
chore(deps): bump github.com/containerd/containerd from 1.6.15 to 1.7.0

Bumps [github.com/containerd/containerd](https://github.com/containerd/containerd) from 1.6.15 to 1.7.0.
- [Release notes](https://github.com/containerd/containerd/releases)
- [Changelog](https://github.com/containerd/containerd/blob/main/RELEASES.md)
- [Commits](https://github.com/containerd/containerd/compare/v1.6.15...v1.7.0)

---
updated-dependencies:
- dependency-name: github.com/containerd/containerd
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dirkmueller committed:
Fix 32bit-x86 typo in testsuite

The GOARCH here is 386 not i386. This caused a slightly odd test
suite failure on that architecture:

--- FAIL: TestPlatformPrepareCommand (0.00s)
     plugin_test.go:45: Expected arg="os-arch", got "linux-s390x"
     plugin_test.go:64: Expected arg="os-arch", got "linux-s390x"

Signed-off-by: Dirk Müller <dirk@dmllr.de>

wujunwei committed:
fix: failed testcase on windows

Signed-off-by: wujunwei <wjw3323@live.com>

amannijhawan committed:
Fixes Readiness Check for statefulsets using partitioned rolling update. (#11774)

* Fixes Readiness Check for statefulsets using partitioned rolling update.
Fixes #11773

This change updates readiness check in ready.go to correctly
account for statefulsets that are utilizing a partitioned upgrade.
These statefulsets only upgrade a subset of the managed pods with each call
to helm upgrade. This causes the upgrade to legitimately hit the condition where
sts.status.CurrentRevision != sts.Status.UpdateRevision which causes helm to mark
the upgrade has failed when in fact it is successful.

This change fixes that behavior to only check when partition is unspecified or 0.

Signed-off-by: Aman Nijhawan <anijhawan@yugabyte.com>

* Adding a unit test to verify that partitioned rolling upgrade for a statefulset works.

Signed-off-by: Aman Nijhawan <anijhawan@yugabyte.com>

---------

Signed-off-by: Aman Nijhawan <anijhawan@yugabyte.com>
Co-authored-by: Aman Nijhawan <anijhawan@yugabyte.com>

dependabot[bot] committed:
chore(deps): bump github.com/docker/docker

Bumps [github.com/docker/docker](https://github.com/docker/docker) from 20.10.21+incompatible to 20.10.24+incompatible.
- [Release notes](https://github.com/docker/docker/releases)
- [Commits](https://github.com/docker/docker/compare/v20.10.21...v20.10.24)

---
updated-dependencies:
- dependency-name: github.com/docker/docker
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

joejulian committed:
create failing test for quietly linting a chart that doesn't exist

Signed-off-by: Joe Julian <me@joejulian.name>

joejulian committed:
fix quiet lint does not fail on non-linting errors

Signed-off-by: Joe Julian <me@joejulian.name>

mattfarina committed:
Merge pull request #11901 from helm/dependabot/go_modules/github.com/containerd/containerd-1.7.0

chore(deps): bump github.com/containerd/containerd from 1.6.15 to 1.7.0

mattfarina committed:
Merge pull request #11966 from helm/dependabot/go_modules/github.com/docker/docker-20.10.24incompatible

chore(deps): bump github.com/docker/docker from 20.10.21+incompatible to 20.10.24+incompatible

mattfarina committed:
Merge pull request #11832 from my-git9/chore-httpstatus

chore:Use http constants as http.request parameters

mattfarina committed:
Merge pull request #11949 from dirkmueller/goarches

Fix 32bit-x86 typo in testsuite

mattfarina committed:
Fix goroutine leak in action install

During the install process there was a place where an install
process could be stuck trying to write to a channel. This would
happen when a context had completed prior to performInstall
finishing. In a short running Helm Client this was not a problem.
But, for long running applications that use Helm as an SDK there
are problems where a memory leak ends up happening due to
goroutines never being able to complete.

This fix provides a means for performInstall to write to its
channel using the method already used to fix the upgrade
issue of the same kind.

Fixes #11805

Signed-off-by: Matt Farina <matt.farina@suse.com>

dependabot[bot] committed:
chore(deps): bump golang.org/x/text from 0.7.0 to 0.9.0

Bumps [golang.org/x/text](https://github.com/golang/text) from 0.7.0 to 0.9.0.
- [Release notes](https://github.com/golang/text/releases)
- [Commits](https://github.com/golang/text/compare/v0.7.0...v0.9.0)

---
updated-dependencies:
- dependency-name: golang.org/x/text
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

mattfarina committed:
Merge pull request #11978 from mattfarina/fix-11805

Fix goroutine leak in action install

mattfarina committed:
Merge pull request #11898 from willzgli/waitbatchPerform

Fix goroutine leak in perform

sabre1041 committed:
Merge pull request #11980 from helm/dependabot/go_modules/golang.org/x/text-0.9.0

chore(deps): bump golang.org/x/text from 0.7.0 to 0.9.0

mattfarina committed:
Only run tests once

Tests are being run twice in CI. Once with code coverage being
report and once without. Tests only need to be run once.

When Helm previously had testing in CircleCI it was the code
coverage tests so that was retained here.

Signed-off-by: Matt Farina <matt.farina@suse.com>

mattfarina committed:
Merge pull request #11985 from mattfarina/only-test-once

Only run tests once

twz123 committed:
Bump k8s.io deps from v0.26.0 to v0.27.0

Adapt to changes in 1.27 client libs:

* The OpenAPIGetter method is now private, but the Validator method does
  what is needed instead.
* The function wait.PollImmediateUntil is deprecated. Switch to the
  recommended new function wait.PollUntilContextCancel.

Related dependency changes:

change k8s.io/kube-openapi v0.0.0-20221012153701-172d655c2280 => v0.0.0-20230308215209-15aac26d736a

And some transitive version changes:

add github.com/go-task/slim-sprig v0.0.0-20210107165309-348f09dbbbc0
add github.com/google/pprof v0.0.0-20210720184732-4bb14d4b1be1
remove github.com/elazarl/goproxy v0.0.0-20180725130230-947c36da3153
github.com/go-errors/errors v1.0.1 => v1.4.2
github.com/go-openapi/jsonpointer v0.19.5 => v0.19.6
github.com/go-openapi/jsonreference v0.20.0 => v0.20.1
github.com/go-openapi/swag v0.19.14 => v0.22.3
github.com/golang/protobuf v1.5.2 => v1.5.3
github.com/mailru/easyjson v0.7.6 => v0.7.7
github.com/onsi/ginkgo/v2 v2.4.0 => v2.9.1
github.com/onsi/gomega v1.23.0 => v1.27.4
github.com/rogpeppe/go-internal v1.9.0 => v1.10.0
golang.org/x/mod v0.8.0 => v0.9.0
golang.org/x/net v0.7.0 => v0.8.0
golang.org/x/term v0.5.0 => v0.6.0
golang.org/x/tools v0.6.0 => v0.7.0
gotest.tools/v3 v3.0.3 => v3.4.0
sigs.k8s.io/json v0.0.0-20220713155537-f223a00ba0e2 => v0.0.0-20221116044647-bc3834ca7abd
sigs.k8s.io/kustomize/api v0.12.1 => v0.13.2
sigs.k8s.io/kustomize/kyaml v0.13.9 => v0.14.1

Signed-off-by: Tom Wieczorek <twieczorek@mirantis.com>

mattfarina committed:
Updating the Helm maintainers

These maintainers have made tremendous contributions but are
currently inactive. I have connected with each of them and they
are focused on other work, at present. They all confirmed moving
to an emeritus status. If they ever want to return to active
development/maintainer-ship of Helm, I would be delighted.

The governance has a process for inactive folks and, as there are
items that need to be voted on, inactive maintainer make decision
making difficult.

These maintainers are and will continue to be missed.

Signed-off-by: Matt Farina <matt.farina@suse.com>

mattfarina committed:
Merge pull request #11479 from MichaelMorrisEst/cascade

Add option to support foreground cascade deletion

yann-soubeyrand committed:
chore: bump k8s.io dependencies to v0.27.1

Signed-off-by: Yann Soubeyrand <yann.soubeyrand@gmx.fr>

mattfarina committed:
Merge pull request #9182 from pscheid92/4030-provide-literal-alternative-for-set-flag

Provide an alternative for --set and/or --set-string to take a value literally

mattfarina committed:
Merge pull request #11973 from joejulian/quiet_lint

fail quiet linting when an error occurs

mattfarina committed:
Merge pull request #11948 from wujunwei/fix-failed-testcase-on-windows

fix: failed testcase on windows

sabre1041 committed:
Attach annotations to OCI artifacts

Signed-off-by: Andrew Block <andy.block@gmail.com>

mattfarina committed:
Merge pull request #12015 from yann-soubeyrand/bump-k8s-io-deps

chore: bump k8s.io dependencies to v0.27.1

mattfarina committed:
Merge pull request #11204 from sabre1041/oci-annotations

Attach annotations to OCI artifacts

dependabot[bot] committed:
chore(deps): bump github.com/Masterminds/semver/v3 from 3.2.0 to 3.2.1

Bumps [github.com/Masterminds/semver/v3](https://github.com/Masterminds/semver) from 3.2.0 to 3.2.1.
- [Release notes](https://github.com/Masterminds/semver/releases)
- [Changelog](https://github.com/Masterminds/semver/blob/master/CHANGELOG.md)
- [Commits](https://github.com/Masterminds/semver/compare/v3.2.0...v3.2.1)

---
updated-dependencies:
- dependency-name: github.com/Masterminds/semver/v3
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

mattfarina committed:
Merge pull request #12034 from mattfarina/update-maintainers

Updating the Helm maintainers

mattfarina committed:
Merge pull request #12039 from helm/dependabot/go_modules/github.com/Masterminds/semver/v3-3.2.1

chore(deps): bump github.com/Masterminds/semver/v3 from 3.2.0 to 3.2.1

dependabot[bot] committed:
chore(deps): bump github.com/Masterminds/squirrel from 1.5.3 to 1.5.4

Bumps [github.com/Masterminds/squirrel](https://github.com/Masterminds/squirrel) from 1.5.3 to 1.5.4.
- [Release notes](https://github.com/Masterminds/squirrel/releases)
- [Commits](https://github.com/Masterminds/squirrel/compare/v1.5.3...v1.5.4)

---
updated-dependencies:
- dependency-name: github.com/Masterminds/squirrel
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

mattfarina committed:
Merge pull request #12042 from helm/dependabot/go_modules/github.com/Masterminds/squirrel-1.5.4

chore(deps): bump github.com/Masterminds/squirrel from 1.5.3 to 1.5.4

joejulian committed:
move Joe Julian to maintainer

Signed-off-by: Joe Julian <me@joejulian.name>

mattfarina committed:
Merge pull request #12059 from joejulian/joe_maintainer

move Joe Julian to maintainer

joejulian committed:
Merge pull request #11870 from cbodonnell/main

full source path for CRDs when output-dir is not provided

KyleFrisbie committed:
Update pkg/action/show_test.go

Co-authored-by: George Jenkins <gvjenkins@gmail.com>
Signed-off-by: Kyle L Frisbie <KyleFrisbie@users.noreply.github.com>

KyleFrisbie committed:
Update pkg/action/show_test.go

Co-authored-by: George Jenkins <gvjenkins@gmail.com>
Signed-off-by: Kyle L Frisbie <KyleFrisbie@users.noreply.github.com>

Commit:
fix test expectation

Commit:
fix test expectation

Commit:
Merge branch 'main' of github.com:KyleFrisbie/helm

joejulian commented:
@KyleFrisbie merging in main messed with this PR which now shows 174 files being changed. Can you please rebase.
